src/Model.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+namespace Pyjac\ORM;
+
+use Doctrine\Common\Inflector\Inflector;
+use PDO;
+use Pyjac\ORM\Exception\ModelNotFoundException;
+
+abstract class Model implements ModelInterface
+{
+    protected $properties = [];
+
+    /**
+     * Store instance of database connection used.
+     *
+     * @var Pyjac\ORM\DatabaseConnection
+     */
+    protected $databaseConnection;
+
+     /**
+      *  The id of the model.
+      *
+      * @property string $id
+      */
+
+     /**
+      * Create a model instance.
+      */
+     public function __construct(DatabaseConnectionInterface $databaseConnection = null)
+     {
+         if ($databaseConnection == null) {
+             $this->databaseConnection = DatabaseConnection::getInstance()->databaseConnection;
+         }
+         $this->databaseConnection = $databaseConnection;
+     }
+
+    /**
+     * Sets into $properties the $key => $value pairs.
+     *
+     * @param string $key
+     * @param string $val
+     */
+    public function __set($key, $val)
+    {
+        $this->properties[$key] = $val;
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return array
+     */
+    public function __get($key)
+    {
+        return $this->properties[$key];
+    }
+
+     /**
+      * Get all the model properties.
+      *
+      * @return array
+      */
+     public function getProperties()
+     {
+         return $this->properties;
+     }
+
+     /**
+      * Set model properties.
+      */
+     public function setProperties(array $properties)
+     {
+         $this->properties = $properties;
+     }
+
+    /**
+     * Pluralize the name of the child class.
+     *
+     * @return string
+     */
+    public function getTableName()
+    {
+        $className = explode('\\', get_called_class());
```

Comment:
I tried your suggested way but I got an error `Only variables should be passed by reference`. This is because the end function requires an argument that can be passed by reference. The means the return value of `explode('\\', get_called_class())` must be stored in a variable before it can then be passed to the `end` function.